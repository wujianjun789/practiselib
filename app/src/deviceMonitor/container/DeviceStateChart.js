/**
 * Created by a on 2017/9/1.
 */
import React,{Component} from 'react';

import Content from '../../components/Content';
export default class DeviceStateChart extends Component{
    constructor(props){
        super(props);
    }

    render(){
        return <Content className="device-state-chart">
            设备状态图
        </Content>
    }
}