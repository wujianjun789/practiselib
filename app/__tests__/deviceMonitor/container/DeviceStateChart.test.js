/**
 * Created by a on 2017/10/9.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

import DeviceStateChart from '../../../src/deviceMonitor/container/DeviceStateChart';

const monitorList = [
    {id:"gateway", name:"网关"},
    {id:"lc", name:"单灯控制器"},
    {id:"screen", name:"显示屏"},
    {id:"collect", name:"数据采集仪"}
]

test('renders Scene', ()=>{
    const component = renderer.create(
        <DeviceStateChart />
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Scene data', ()=>{
    const component = shallow(<DeviceStateChart />);
    component.setState({ monitorList:monitorList});
    expect(component.find('.monitor').length).toEqual(monitorList.length);
})