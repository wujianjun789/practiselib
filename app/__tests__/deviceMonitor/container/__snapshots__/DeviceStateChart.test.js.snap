exports[`test renders Scene 1`] = `
<div
  className="content device-state-chart">
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      网关
    </div>
    <div
      className="body statistics-container">
      <div
        className="progress-container">
        <span
          className="title">
          故障/总数
        </span>
        <div
          className="progress fault">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={30}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "10%",
              }
            }>
            30/300
          </div>
        </div>
      </div>
      <div
        className="progress-container">
        <span
          className="title">
          在线/总数
        </span>
        <div
          className="progress online">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={120}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "40%",
              }
            }>
            120/300
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer">
      <span
        className="left">
        0
      </span>
      <span
        className="right">
        100%
      </span>
    </div>
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      单灯控制器
    </div>
    <div
      className="body statistics-container">
      <div
        className="progress-container">
        <span
          className="title">
          故障/总数
        </span>
        <div
          className="progress fault">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={30}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "10%",
              }
            }>
            30/300
          </div>
        </div>
      </div>
      <div
        className="progress-container">
        <span
          className="title">
          在线/总数
        </span>
        <div
          className="progress online">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={120}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "40%",
              }
            }>
            120/300
          </div>
        </div>
      </div>
      <div
        className="progress-container">
        <span
          className="title">
          亮灯率
        </span>
        <div
          className="progress lightRate">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={150}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "50%",
              }
            }>
            50%
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer">
      <span
        className="left">
        0
      </span>
      <span
        className="right">
        100%
      </span>
    </div>
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      显示屏
    </div>
    <div
      className="body statistics-container">
      <div
        className="progress-container">
        <span
          className="title">
          故障/总数
        </span>
        <div
          className="progress fault">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={30}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "10%",
              }
            }>
            30/300
          </div>
        </div>
      </div>
      <div
        className="progress-container">
        <span
          className="title">
          在线/总数
        </span>
        <div
          className="progress online">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={120}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "40%",
              }
            }>
            120/300
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer">
      <span
        className="left">
        0
      </span>
      <span
        className="right">
        100%
      </span>
    </div>
  </div>
  <div
    className="col-sm-6 monitor">
    <div
      className="heading">
      数据采集仪
    </div>
    <div
      className="body statistics-container">
      <div
        className="progress-container">
        <span
          className="title">
          故障/总数
        </span>
        <div
          className="progress fault">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={30}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "10%",
              }
            }>
            30/300
          </div>
        </div>
      </div>
      <div
        className="progress-container">
        <span
          className="title">
          在线/总数
        </span>
        <div
          className="progress online">
          <div
            aria-valuemax={300}
            aria-valuemin={0}
            aria-valuenow={120}
            className="progress-bar "
            role="progressbar"
            style={
              Object {
                "width": "40%",
              }
            }>
            120/300
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer">
      <span
        className="left">
        0
      </span>
      <span
        className="right">
        100%
      </span>
    </div>
  </div>
</div>
`;
